dialogue-clients/src/main/java/com/palantir/dialogue/clients/DefaultDialogueDnsResolver.java

```diff
@@ -41,8 +45,60 @@ public ImmutableSet<InetAddress> resolve(String hostname) {
             }
             return ImmutableSet.copyOf(results);
         } catch (UnknownHostException e) {
-            log.warn("Unknown host '{}'", UnsafeArg.of("hostname", hostname), e);
+            ExtractedGaiError gaiError = extractGaiErrorString(e);
+            log.warn(
+                    "Unknown host '{}'",
+                    SafeArg.of("gaiErrorMessage", gaiError.getErrorMessage()),
+                    SafeArg.of("gaiErrorType", gaiError.getErrorType()),
```

Comment:
Thoughts on adding a `meter` metric tagged with the `errorType` so we can keep an eye on things more broadly?

Reply:
yeah i think that makes sense, will add